UI/src/app/components/comfy/comfy.component.ts

```diff
@@ -407,15 +408,18 @@ export class ComfyComponent extends BaseComponent implements OnInit, AfterViewIn
     if (idx > -1) {
       switch (area) {
         case 'source': {
-          this.source = this.source.filter(filterByName);
+          this.source = this.data.source.map(item => item.name).filter(filterByName);
           break;
         }
         case 'target': {
           this.targetTableNames = uniq(Object.keys(this.targetConfig)).filter(filterByName);
           break;
         }
         case 'source-column': {
-          this.sourceRows = this.sourceRows.filter(row => filterByName(row.name));
+          this.sourceRows = uniqBy(this.data.source.
+          map(table => table.rows).
```

Comment:
you don't need to have three operations there. You already use reduce, you can get rows for table and apply filtering inside of it

Reply:
Done